src/python/WMCore/Services/LogDB/LogDB.py

```diff
@@ -126,5 +127,55 @@ def cleanup(self, thr, backend='local'):
         try:
             self.backend.cleanup(thr)
         except Exception as exc:
-            self.logger.error('LogDBBackend cleanup API failed, backend=%s, error=%s' \
-                    % (backend, str(exc)))
+            self.logger.error('LogDBBackend cleanup API failed, backend=%s, error=%s', backend, str(exc))
+    
+    def heartbeat_report(self):
+        report = defaultdict(dict)
+        if self.user_pat.match(self.identifier):
+            self.logger.error("User %s: doesn't allow this function", self.identifier) 
+            return report
+        
+        for row in self.backend.get(self.default_user, None, agent=True).get('rows', []):
+            identifier = row['doc']['identifier']
+            # wmstats thread can run in multiple boxed. 
+            if self.identifier == identifier or identifier.startswith(self.identifier):
+                #this will handle wmstats DataCacheUpdate thread with multiple machine
+                postfix = identifier.replace(self.identifier, "")
```

Comment:
I think I got lost in this review. First, why checking if identifier startswith ... ? Isn't it supposed to be an unique key (cherrypy thread names, agent names, request names)?
What's the difference between identifier and doc.thr? This should tell me what is postfix for.